tez-dag/src/main/java/org/apache/tez/dag/app/DAGAppMaster.java

```diff
@@ -952,6 +960,12 @@ public void shutdown(boolean now) {
       }
       LOG.info("Handling DAGAppMaster shutdown");
 
+      // Check if the thread dump service is up in any case, if yes attempt a shutdown
+      if (tezThreadDumpHelper != null) {
```

Comment:
I don't think we need this here, but if you want to be cautious, you can still do this into serviceStop